src/async-wrap.cc

```diff
@@ -617,10 +637,29 @@ AsyncWrap::AsyncWrap(Environment* env,
 
   // Use AsyncReset() call to execute the init() callbacks.
   AsyncReset(silent);
+
+  switch (provider_type()) {
+#define V(PROVIDER)                                                           \
+    case PROVIDER_ ## PROVIDER:                                               \
+      TRACE_EVENT_NESTABLE_ASYNC_BEGIN0("node", #PROVIDER,                    \
+        static_cast<int64_t>(get_id()));                                      \
+      break;
+    NODE_ASYNC_PROVIDER_TYPES(V)
+#undef V
+  }
 }
 
 
 AsyncWrap::~AsyncWrap() {
+  switch (provider_type()) {
+#define V(PROVIDER)                                                           \
```

Comment:
> Just to make sure I understand, currently, no trace events will be reported for httpparser

You will get an `init()` every time an http `node::Parser` is initialized, and you'll get `before()`/`after()` for that one as well, but you won't get a `destroy()` for that parser if it's placed in the `freeParser()` pool.

When the `Parser` is reused a new `async_id` is assigned, then you'll get a `before()`/`after()` call for the `async_id` that hasn't had `init()` called yet. Creating a broken call graph.

The `init()` call can be fixed by placing `TRACE_EVENT_NESTABLE_ASYNC_BEGIN0` in `node::Parser::Reinitialize()`, but there's no way to track when the `node::Parser` instance has been placed in the pool from C++. So it would require a JS call.

The `freeParser()` code path isn't hot, so it's not unreasonable to put a JS call in there to notify C++ that the parser is done and it can emit `destroy()`.